src/daemonprivate.cpp

```diff
@@ -70,6 +70,20 @@ DaemonPrivate::DaemonPrivate(Daemon* parent)
     });
 
     getAllProperties();
+
+    // If PackageKit is not running, we are not getting a signal when
+    // properties are updated (like when the network changes its state)
+    // so we connect to the NetworkManager dbus signal directly and then
+    // get all properties again to force packagekit to run and
+    // update the properties so we can emit the appropiate signals.
+    q_ptr->connect(&networkManagerMonitor, SIGNAL(networkStateChanged(uint)),
+                   SLOT(getAllPropertiesIfPackageKitNotRunning()));
+}
+
+void DaemonPrivate::getAllPropertiesIfPackageKitNotRunning()
+{
+    if (!running)
+        getAllProperties();
```

Comment:
If PK is not running I believe getAllProperties() is likely to wake it up again, thus, talking to NM useless if you don't let it stop.

Reply:
Well, that's the whole point of this PR. We talk to NM because, when PK is not running, it doesn't emit the [propertiesChanged signal](https://github.com/hughsie/PackageKit-Qt/blob/9652f94119523ab1125b3be435c367913e18869b/src/daemon.cpp#L55) when the network state changes, so we use NM to have an alternate way to find out we need to update PK properties while PK is not running. Once we got notified the network is up, we call getAllPropertiesIfPackageKitNotRunning, which, as you said, wakes PK so it can give us the updated properties through the usual channels.

Reply:
Note that the check for [`!running`](https://github.com/hughsie/PackageKit-Qt/pull/30/commits/9652f94119523ab1125b3be435c367913e18869b#diff-c077728eae8e702f1ce3420e677f5f21R85) is important since that controls that we only do something different than without this PR when PK is stopped.